### DISCUSSION TITLE
Need Image upgrade script for Checkpoint

### DISCUSSION THREAD

**Discussion by sen2diwakar** [ktbyers/netmiko#3409] [General]
Hi KT sir and all other gentlemen, 

Greetings !

Need a help from you .  Kindly provide   Checkpoint IOS upgrade script  through Netmiko via cli . I have worked on 1000+ scripts till date on F5 , Cisco and Fortigate . However I am new to CheckPoint and do not have knowledge on upgrade procedure hence absolutely confused.

Please if anyone can help I will be grateful . Here is a sample script i made for Checkpoint as working : 

```
import re
from netmiko import Netmiko
import time




def connect_cp():
    try:
        switch = {
            'device_type': 'checkpoint_gaia', 'ip': 'ip_address', 'username':
            'username', 'password':"password", "conn_timeout": 2}
        switch_connect = Netmiko(**switch)
        print("ip" + " " + "is reachable")
        prompt_sw_axa = switch_connect.find_prompt()
        print(prompt_sw_axa)

    except Exception as e:
        print (e)

    output = switch_connect.send_command ("show interfaces all")
    print(output)
    #LINES = output.split("\n")
    #print(LINES)

    switch_connect.disconnect()



connect_cp()
```

**Comment by ktbyers** [comment]
Yeah, I will pass on this...not an appropriate question / discussion item

**Reply by sen2diwakar** [reply]
Hi  kt,

Any guidance here is higly aprreciated . No one has helped me out so far on this question
